x-pack/plugin/relevance-search/src/test/java/org/elasticsearch/xpack/relevancesearch/relevance/RelevanceSettingsServiceTests.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.relevance;
+
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.test.ESSingleNodeTestCase;
+import org.elasticsearch.xpack.relevancesearch.relevance.boosts.FunctionalBoost;
+import org.elasticsearch.xpack.relevancesearch.relevance.boosts.ProximityBoost;
+import org.elasticsearch.xpack.relevancesearch.relevance.boosts.ScriptScoreBoost;
+import org.elasticsearch.xpack.relevancesearch.relevance.boosts.ValueBoost;
+import org.elasticsearch.xpack.relevancesearch.relevance.settings.RelevanceSettings;
+import org.elasticsearch.xpack.relevancesearch.relevance.settings.RelevanceSettingsService;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class RelevanceSettingsServiceTests extends ESSingleNodeTestCase {
+
+    private RelevanceSettingsService service;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        service = getInstanceFromNode(RelevanceSettingsService.class);
+        createIndex(RelevanceSettingsService.ENT_SEARCH_INDEX);
+    }
+
+    public void testParseFields() throws Exception {
+        String settingsId = "1";
+        indexDocument(
+            RelevanceSettingsService.ENT_SEARCH_INDEX,
+            RelevanceSettingsService.RELEVANCE_SETTINGS_PREFIX + settingsId,
+            Map.of("query_configuration", Map.of("fields", List.of("title^3", "description^2")))
+        );
+        RelevanceSettings settings = service.getRelevanceSettings(settingsId);
+        Map<String, Float> expected = Map.of("title", 3f, "description", 2f);
+        assertEquals(expected, settings.getQueryConfiguration().getFieldsAndBoosts());
+    }
+
+    public void testParseValueBoost() throws Exception {
```

Comment:
Would it be worth testing multiple boosts?

Reply:
We don't need this for a demo, but I'll add some if I have the time.